src/opnsense/mvc/app/models/OPNsense/Routing/Migrations/M1_0_0.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * Copyright (C) 2023 Deciso B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\Routing\Migrations;
+
+use OPNsense\Base\BaseModelMigration;
+use OPNsense\Core\Config;
+use OPNsense\Routing\Gateways;
+
+class M1_0_0 extends BaseModelMigration
+{
+    /**
+     * Migrate older models into shared model
+     * @param $model
+     */
+    public function run($model)
+    {
+        $config = Config::getInstance()->object();
+
+        if (!empty($config->gateways) && !empty($config->gateways->gateway_item)) {
+            foreach ($config->gateways->gateway_item as $gateway) {
+                $node = $model->gateway_item->Add();
+
+                // monitor_disable was "on" when no node present
+                $node->monitor_disable = !empty((string)$gateway->monitor_disable) ? '1' : '0';
+
+                if (empty((string)$gateway->priority)) {
+                    $node->priority = '255';
+                }
+
+                if (empty((string)$gateway->ipprotocol)) {
+                    $node->ipprotocol = 'inet';
+                }
+
+                foreach ($gateway as $key => $value) {
+                    if ($key === 'gateway') {
+                        // change all occurences of "dynamic" to empty string
+                        $node->gateway = str_replace('dynamic', '', (string)$value);
+                        continue;
+                    }
+
+                    if ($key === 'alert_interval') {
```

Comment:
can't we match this against model nodes?

Reply:
how about https://github.com/opnsense/core/pull/6808/commits/373a48c274b6a2f1ed1f3a42e8ee37ae70181e05?